vignettes/datatable-programming.Rmd

```diff
@@ -373,6 +373,29 @@ print(j)
 DT[, j, env = list(j = j)]
 ```
 
+### Injecting functions using strings in `env`
+
+In `data.table`, you can inject functions into your expressions by passing their names as strings in the `env` parameter. This method allows you to use function names directly as strings, and `data.table` will automatically interpret these strings as the corresponding functions. This approach simplifies the process and makes it easier to work with function names dynamically.
+
+Suppose you want to calculate the total of `Sepal.Length` in the `iris` dataset using the `sum` function. You can inject the `sum` function by passing its name as a string in the `env` parameter.
+
+```{r sum_example}
+result <- DT[, f(Sepal.Length), env = list(f = "sum")]
```

Comment:
IMO other pieces of code in this vignette are sufficiently covering parametrizing function names

Reply:
so do we need addition to vignette or not, tbf while adding i also felt that eveything is already covered so i added a section for  specific case mentioned in the issue. 

> Inconsistency is when someone will try to actually inject function body via passing function to env argument. Don't have reasonable use case for that but I don't think we can rule out such situation. For env it is actually sufficient to provide `"sum"` because it is automatically turned into name. So not even need to quote() or as.name(). What seems very reasonable is to add this use case to examples so it is documented to use `"sum"` rather than `sum`.